File: IClient.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Threading;
using System.Threading.Tasks;
using FlipDriveSDK.JSON;

namespace FlipDriveSDK
{
	public interface IClient
	{
		Task<JSON_UserInfo> UserInfo();

		Task<JSON_List> List(string DestinationFolderID, utilitiez.FilterEnum Filter, utilitiez.SortEnum Sort, int Limit = 1000);

		Task<JSON_Item> Upload(object FileToUpload, utilitiez.UploadTypes UploadType, string DestinationFolderID, string FileName, IProgress<ReportStatus> ReportCls = null, CancellationToken token = default(CancellationToken));

		Task<JSON_Item> CreateNewFolder(string DestinationFolderID, string FolderName);

		Task<bool> Rename(string DestinationID, string NewName);

		Task<bool> Move(List<string> SourceIDs, string DestinationFolderID);

		Task<bool> Trash(List<string> DestinationIDs);

		Task<bool> Delete(List<string> DestinationIDs);

		Task<Stream> DownloadAsStream(string DestinationFileID, IProgress<ReportStatus> ReportCls = null, CancellationToken token = default(CancellationToken));

		Task Download(string DestinationFileID, string FileSaveDir, string FileName, IProgress<ReportStatus> ReportCls = null, CancellationToken token = default(CancellationToken));

		Task<Stream> FileThumbnail(string DestinationFileID, FClient.Size ThumbSize, IProgress<ReportStatus> ReportCls = null, CancellationToken token = default(CancellationToken));

		Task<bool> AddToFavorite(List<string> DestinationIDs);

		Task<bool> RemoveFromFavorite(List<string> DestinationIDs);

		Task<JSON_ListRecents> ListRecents(int Limit = 250);

		Task<JSON_List> ListFavorites(utilitiez.FilterEnum Filter, utilitiez.SortEnum Sort, int Limit = 1000);

		Task<JSON_List> ListRecycleBin(utilitiez.FilterEnum Filter, utilitiez.SortEnum Sort, int Limit = 1000);

		Task<bool> EmptyRecycleBin();

		Task<JSON_List> Search(string Keyword, utilitiez.FilterEnum Filter, utilitiez.SortEnum Sort, int Limit = 1000);
	}
}

File: src/FlipDriveSDK/FlipDriveSDK/IClient.vb
Imports FlipDriveSDK.JSON
Imports FlipDriveSDK.utilitiez

Public Interface IClient

    ''http://api2.flipdrive.com/4.0/share/public/download?api_key=web&code=K_BCd5G
    ''https://api.flipdrive.com/2.0/rest

    ReadOnly Property File(FileID As String) As IFile
    ReadOnly Property File As IFile
    ReadOnly Property Folder(FolderID As String) As IFolder
    ReadOnly Property Folder As IFolder

    Function DriveInfo(Drive As DriveTypeEnum) As Task(Of JSON_DriveMetadata)
    Function UserInfo() As Task(Of JSON_UserInfo)
    Function ListRecents(Optional Limit As Integer = 250) As Task(Of JSON_ListRecents)
    Function ListFavorites(Filter As utilitiez.FilterEnum, Sort As utilitiez.SortEnum, Optional Limit As Integer = 1000, Optional OffSet As Integer = 0) As Task(Of JSON_FolderList)
    Function ListRecycleBin(Filter As utilitiez.FilterEnum, Sort As utilitiez.SortEnum, Optional Limit As Integer = 1000, Optional OffSet As Integer = 0) As Task(Of JSON_FolderList)
    Function EmptyRecycleBin() As Task(Of Boolean)
    Function Search(Keyword As String, Drive As DriveTypeEnum, Filter As utilitiez.FilterEnum, Sort As utilitiez.SortEnum, Optional Limit As Integer = 1000, Optional OffSet As Integer = 0) As Task(Of JSON_FolderList)

End Interface
